pkg/mgmt/backup/doc.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+The Backup flow is as follows:
+
+- plugin takes the backup of ZFSVolume CR so that it can be restored.
+
+- It will save the namespace information where the pvc is created also while taking the backup. Plugin will use this info if restoring without a namespace mapping to find if volume has already been restored.
+
+- plugin then creates the ZFSBackup CR with the destination volume and remote location where the data needs to be send.
+
+- Backup controller (on node) keeps a watch for new CRs associated with the node id. This node ID will be same as the Node ID present in the ZFSVolume resource.
+
+- The Backup controller will take a snapshot and then send the data to the remote location.
```

Comment:
better to remove `send the data to remote location` since it is part of next step and may create confusion.